src/store/messagesStore.js

```diff
@@ -861,10 +862,12 @@ const actions = {
 			return
 		}
 
+		const idToUpdate = (id === null) ? conversation.lastMessage.id : id
```

Comment:
For fed conversations; there is no `conversation.lastMessage.id`, so if it is null, it should stay `null` ?

Reply:
and the capability `markAsReadWithoutLast`, was it added for feds only or for all conversations in v19 ?

Reply:
It's applied since v19.
`idToUpdate` needed only for immediate frontend handling, and for federated ones, null wouldn't be correct. Best approach would be to get last id from messagesList

Reply:
ok but that will render an error of undefined for federated conversations, should it get the id of the last message in `messagesList` ?

Reply:
>  Best approach would be to get last id from messagesList

agree

Reply:
As it's only a visual change, I've decided to skip one step for federated conversations.
- LeftSidebar will be updated anyway after server response received
- Visual marker will be updated, only if chat is loaded -> last message id is known